app/src/main/java/com/example/hepiplant/EventEditActivity.java

```diff
@@ -0,0 +1,208 @@
+            @Override
+            public void onClick(View v) {
+                if(eventName.getText()!=null && !eventName.getText().toString().equals(" ")) patchEventResponse();
+                else Toast.makeText(getApplicationContext(),"Podaj tytuł wydarzenia",Toast.LENGTH_LONG).show();
+            }
+        });
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        Log.v(TAG, "onActivityResult");
+        super.onActivityResult(requestCode, resultCode, data);
+        if (requestCode == 1) {
+            if (resultCode == RESULT_OK) {
+                Log.v(TAG, data.getExtras().getString("data"));
+                eventDate.setText(data.getExtras().getString("data"));
+                Log.v(TAG, eventDate.getText().toString());
+            }
+        }
+    }
+
+    private void patchEventResponse(){
+        JSONObject postData = new JSONObject();
+        try {
+            postData.put("eventName",eventName.getText().toString());
+            if(eventDate.getText().toString().contains(":"))
+                postData.put("eventDate",eventDate.getText().toString());
+            else
+                postData.put("eventDate",eventDate.getText().toString().trim()+" 00:00:00");
+            postData.put("eventDescription",eventDescription.getText().toString());
+            postData.put("done",false);
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+        Log.v(TAG,"patch");
+        RequestQueue queue = Volley.newRequestQueue(getApplicationContext());
+        String url = getRequestUrl()+getIntent().getExtras().getLong("eventId");
+        Log.v(TAG,url);
+        Log.v(TAG,postData.toString());
+        JsonObjectRequest jsonArrayRequest = new JsonObjectRequest(Request.Method.PATCH, url, postData,
+                new Response.Listener<JSONObject>() {
+                    @RequiresApi(api = Build.VERSION_CODES.N)
+                    @Override
+                    public void onResponse(JSONObject response) {
+                        Log.v(TAG,"onResponse");
+                        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                            onPatchResponseEvent(response);
+                        }
+                    }
+                }, new Response.ErrorListener() {
+            @Override
+            public void onErrorResponse(VolleyError error) {
+                onErrorResponse(error);
+            }
+        }){
+            @Override
+            public Map<String, String> getHeaders() {
+                return prepareRequestHeaders();
+            }
+        };
+
+        queue.add(jsonArrayRequest);
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.O)
+    private void onPatchResponseEvent(JSONObject response){
+        Log.v(TAG, "ONResponse");
+        String str = String.valueOf(response); //http request
+        EventDto data = new EventDto();
+        Gson gson = new Gson();
```

Comment:
Mamy gsona w configu, nie trzeba robić nowego. Ale nie poprawiaj, zaraz to będzie zrobione używając JSONResponseHandler